### ISSUE TITLE
Allow overlays over dependencies

### ISSUE THREAD

**Issue by juanitocalero** [akhikhl/gretty#309]
It would be great if Gretty could make overlays over WAR dependencies, no only on other projects from the build. That would be really useful.

Maybe it could benefit also from the composite builds feature since Gradle 3.1.

I have a complex build environment with custom "child" webapps in different repositories, and I'll be very interested in such a feature. Moreover, if the requirements are not complicated and if the project owners could give some hints or help about the implementation, my team and I could try to develop it.

**Comment by saladinkzn** [comment]
Did you try https://github.com/saladinkzn/gretty/issues/4?

I've imlemented this feature in my fork and it made it to gretty - 1.4.0

**Comment by akhikhl** [comment]
Assuming the issue is resolved

**Comment by juanitocalero** [comment]
Well, the issue is now closed, but I finally found the time to test @saladinkzn suggestion, and it works, but I found some issues:
- Dependency overlays are only available inside farms, and that's a little counterintuitive
- The feature is not documented, and I spent quite some time to find the right syntax (the highly dynamic nature of Groovy and Gradle doesn't help here)
- The overlay mode is only triggered if the webappref is a file or dependency. My use case is the other way around. I want to launch the "child" webapp in its project, and get the "base" webapp from a dependency
I want this:
```
farm {
	webapp project, overlays: ['com.company:base-webapp:1.0.0-SNAPSHOT']
}
```

But the overlay only works from my base app build.gradle:
```
farm {
	webapp 'com.company:child-webapp:1.0.0-SNAPSHOT', overlays: [':base']
}
```
It would be great if this dependency overlay feature could be more tightly integrated in the main Gretty overlay feature. Anyway, thanks for the good work!

**Comment by danielabutano** [comment]
Hi @akhikhl and @saladinkzn,
some weeks ago I started migrating the intermine build system (which has a complex build system based on ant) to gradle.
I have partially  migrated the intermine core ( gradle multiproject containing as sub project also intermine-webapp); now I've started working on migrating the webapp instances  built over the "intermine-webapp" which include testmodel webapp, flymine, humanmine...
I started from testmodel webapp.
I would really like to be able to configure in the testmodel webapp (as also in humanmine, etc) something like that:
``
gretty {
  //overlay 'org.intermine:intermine-webapp:2.0.0-SNAPSHOT'
}
``
Here my efforts: https://github.com/danielabutano/intermine/tree/gradle-with-artifact/testmodel/webapp
There is any chance this might work?
As the comment above,  I would really like to launch the "child" webapp (testmodel webapp) in its project, and get the "base" webapp (intermine-webapp) from a dependency.
Any suggestions?
Thanks a lot!
